### ISSUE TITLE
Consolidate how we set default tag and allowed tags for accurate documentation

### ISSUE THREAD

**Issue by khiga8** [primer/view_components#491]
In [System arguments](https://primer.style/view-components/system-arguments), we have: 
`All Primer ViewComponents accept a standard set of options called system arguments`. This includes a `tag` option but it isn't necessarily true that a consumer can update tags. In reality, some of our components don't allow tags to be change since we set internally:
```
system_arguments[:tag] = :input
```

Other times, we have a default and allow tag to be changed based on user input:
```
system_arguments[:tag] ||= :input
```
OR
```
 def initialize(
      tag: DEFAULT_TAG,
```

And other times, we have a default tag which can be changed to limited tag options.
```
@system_arguments[:tag] = fetch_or_fallback(TAG_OPTIONS, tag, DEFAULT_TAG)
```

The documentation for components don't always surface this information making it unclear what the default tag is, what tags are allowed, etc. 

Our `Rake` populates the default tag section by looking the component initializer and checking if there's a tag keyword: 
```
 def initialize(
      tag: DEFAULT_TAG,
```
However, we don't always set default tag like this so that section in the doc ends up being`N/A` for most components. This is a problem because it is ambiguous without looking at the component source code what tags are allowed, what the default tag is, etc. 

TODOS:

- [x] Remove tag from system arguments docs since it's misleading
- [ ] Create tag allow list (currently known as TAG_OPTIONS) for the components and slots that do allow restricted set of tags. These allowed tags should be surfaced in the docs in Arguments table. Looks like we do this for a few components already like Button.
- [ ] For components that can render any tag, declare a tag parameter with a default. 
- [ ] For components that have fixed tags that can't be changed, no need to document in Arguments though we may mention what the tag is if it's important.
- [ ] Maybe.... update `Contributing` docs?

**Comment by joelhawksley** [comment]
@khiga8 off the top of my head, I _think_ we should be explicitly allow listing tags for every component.

**Comment by khiga8** [comment]
```
# remove tag from system arguments docs 
# create TAG_ALLOWLIST for each component and slot we want to limit 
# when tag has fixed tag, don't need to surface unless we want to 
# for components that can render any tag, declare a tag parameter with default 
```

**Comment by khiga8** [comment]
TO REVIEW - components / slots that allow flexible tag

- [ ] avatar_stack_component 
- [ ] avatar_stack_component `body_arguments` 
- [ ] blankslate_component
- [ ] button_group
- [ ] details_component `:body` slot
- [ ] hidden_text_expander
- [ ] label_component
- [ ] markdown_component
- [ ] menu_component `:heading` slot
- [ ] popover_component
- [ ] popover_component `:heading` slot
- [ ] subhead_component `:heading` slot
- [ ] tab_nav_component
- [ ] text_component
- [ ] tooltip_component
- [ ] truncate
- [ ] underline_nav_component `:actions` slot
- [ ] navigation/tab_component `:panel` slot

**Comment by khiga8** [comment]
6/21/21 Component-driven UI meeting

- Leave out `Tooltip` because the `Tooltip` API itself is questionable. Debate about whether it should be a component.

**Comment by joelhawksley** [comment]
@khiga8 I still think we should have an allow list of some sort for `Tooltip`. Do you think there's a minimal approach we could take to land _something_, even if it's not final?

**Comment by khiga8** [comment]
I chatted with `@manuelpuyol` in my pairing session last week about what the allowed tags for `Tooltip` should be. The conversation shifted and both of us reached the conclusion that `Tooltip` is odd for a number of reasons and we should direct our efforts to either deprecating it or converting it to `system_arguments`. `Tooltip` will be out of scope of this issue, but I've opened an issue #663 to follow up.
